### PR TITLE
Refs [TASK][41201] Create remote data source

### PR DESCRIPTION
**Pull request by vuphong0x**
## Related Tickets
- [#Task 41201: Create remote data source](https://edu-redmine.sun-asterisk.vn/issues/41201)
## WHAT
- 
## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://github.com/framgia/coding-standards/tree/master/eng/android |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Conventions | Kotlin coding conventions | https://kotlinlang.org/docs/coding-conventions.html |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/sun/efood/data/repository/MealCategoryRepository.kt [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package com.sun.efood.data.repository
+
+import com.sun.efood.data.model.MealCategory
+import com.sun.efood.data.source.MealCategoryDataSource
+import com.sun.efood.data.source.utils.OnDataCallback
+
+class MealCategoryRepository constructor(
```
Comment (thonc-2135) [inline app/src/main/java/com/sun/efood/data/repository/MealCategoryRepository.kt]: private constructor chứ, ms comment pull trc mà nhỉ

File: app/src/main/java/com/sun/efood/data/repository/MealCategoryRepository.kt [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    private val remote: MealCategoryDataSource
+) : MealCategoryDataSource {
```
Comment (ghost) [inline app/src/main/java/com/sun/efood/data/repository/MealCategoryRepository.kt]: để private constructer đi e

File: app/src/main/java/com/sun/efood/data/repository/MealRepository.kt [hunk 3]
```diff
@@ -0,0 +1,34 @@
+package com.sun.efood.data.repository
+
+import com.sun.efood.data.model.Meal
+import com.sun.efood.data.model.MealDetail
+import com.sun.efood.data.source.MealDataSource
+import com.sun.efood.data.source.utils.OnDataCallback
+
+class MealRepository constructor(
+    private val remote: MealDataSource.Remote
+) : MealDataSource.Remote {
```
Comment (ghost) [inline app/src/main/java/com/sun/efood/data/repository/MealRepository.kt]: nữa này

**Review by vietnh-1109 — APPROVED** [review]

**Review by huytq-2422 — APPROVED** [review]

File: app/src/main/java/com/sun/efood/utils/FragmentUtil.kt [hunk 4]
```diff
@@ -0,0 +1,50 @@
+package com.sun.efood.utils
+
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.FragmentManager
+import androidx.fragment.app.FragmentTransaction
+
+object FragmentUtil {
+
+    fun addFragment(
+        fragmentManager: FragmentManager,
```
Comment (huytq-2422) [inline app/src/main/java/com/sun/efood/utils/FragmentUtil.kt]: Em chuyển sang viết dưới dạng Extension 
Extension sẽ gen cho em 1 class tương tự class Util của em trong Java 
Chú ý viết ở top-level, không viết trong object hay class 
```suggestion
fun FragmentManager.addFragment(
```

File: app/src/main/java/com/sun/efood/utils/FragmentUtil.kt [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        fragment: Fragment,
+        tag: String? = null
+    ) {
+        fragmentManager.beginTransaction()
+            .apply {
+                setTransition(FragmentTransaction.TRANSIT_FRAGMENT_OPEN)
+                add(layoutId, fragment, tag)
+                addToBackStack(tag)
+                commit()
+            }
```
Comment (huytq-2422) [inline app/src/main/java/com/sun/efood/utils/FragmentUtil.kt]: anh nhớ k nhầm thì fragment manager đã viết dưới dạng builder pattern rồi mà nhỉ, nên không cần đặt trong scope function apply nữa 
```suggestion
        fragmentManager.beginTransaction()
            .setTransition(FragmentTransaction.TRANSIT_FRAGMENT_OPEN)
            .add(layoutId, fragment, tag)
                addToBackStack(tag)
                commit()
           
```

**Review by thonc-2135 — APPROVED** [review]

**Pull request merged**